system-int-test/test/cmr/system_int_test/search/subscription/subscription_search_test.clj

```diff
@@ -202,6 +202,72 @@
             (search/find-concepts-in-format
              nil :subscription {} {:url-extension "atom"}))))))
 
+(deftest search-for-subscription-by-type-test
+  (let [_ (mock-urs/create-users (system/context) [{:username "SubId1" :password "Password"}
+                                                   {:username "SubId2" :password "Password"}])
+        coll1 (data2-core/ingest-umm-spec-collection
+               "PROV4"
+               (data-umm-c/collection
+                {:ShortName "coll1"
+                 :EntryTitle "entry-title1"})
+               {:token "mock-echo-system-token"})
+        coll2 (data2-core/ingest-umm-spec-collection
+               "PROV4"
+               (data-umm-c/collection
+                {:ShortName "coll2"
+                 :EntryTitle "entry-title2"})
+               {:token "mock-echo-system-token"})
+        subscription1 (subscriptions/ingest-subscription-with-attrs {:native-id "sub1"
+                                                                     :Name "Subscription1"
+                                                                     :SubscriberId "SubId1"
+                                                                     :Query "platform=NOAA-6"
+                                                                     :CollectionConceptId (:concept-id coll1)
+                                                                     :provider-id "PROV1"})
+        subscription2 (subscriptions/ingest-subscription-with-attrs {:native-id "sub2"
+                                                                     :Name "Subscription2"
+                                                                     :SubscriberId "SubId2"
+                                                                     :Query "platform=NOAA-7"
+                                                                     :CollectionConceptId (:concept-id coll2)
+                                                                     :provider-id "PROV2"})
+        subscription3 (subscriptions/ingest-subscription-with-attrs {:native-id "sub3"
+                                                                     :Name "Subscription3"
+                                                                     :SubscriberId "SubId1"
+                                                                     :Type "collection"
+                                                                     :Query "platform=NOAA-10"
+                                                                     :CollectionConceptId (:concept-id coll1)
```

Comment:
Remove this line. Collection subscription should not have `CollectionConceptId` field.